webrender_traits/src/types.rs

```diff
@@ -16,6 +16,15 @@ use std::sync::Arc;
 #[cfg(target_os = "macos")] use core_graphics::font::CGFont;
 #[cfg(target_os = "windows")] use dwrote::FontDescriptor;
 
+// Some stubs to work around https://github.com/serde-rs/serde/issues/720
+#[cfg(not(feature = "webgl"))]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GLContextAttributes([u8; 0]);
```

Comment:
Apparently it's the sound way of making opaque enums that can't be instantiated, see https://github.com/rust-lang/rfcs/pull/1861#issuecomment-274613536